pkg/broker/broker_test.go

```diff
@@ -17,24 +17,211 @@
+					ID:   u,
+					Spec: &bundle.Spec{},
+				}, nil)
+			},
+		},
+		{
+			name: "get service instance with dashboard url",
+			dao:  new(mocks.Dao),
+			config: Config{
+				DashboardRedirector: "url.com",
+			},
+			addExpectations: func(d *mocks.Dao) {
+				d.On("GetServiceInstance", u.String()).Return(&bundle.ServiceInstance{
+					ID: u,
+					Spec: &bundle.Spec{
+						Alpha: map[string]interface{}{
+							"dashboard_redirect": true,
+						},
+					},
+				}, nil)
+			},
+			dashboardURL: fmt.Sprintf("http://%v/?id=%v", "url.com", u.String()),
+		},
+		{
+			name: "get service instance with dashboard url and https",
+			dao:  new(mocks.Dao),
+			config: Config{
+				DashboardRedirector: "https://url.com",
+			},
+			addExpectations: func(d *mocks.Dao) {
+				d.On("GetServiceInstance", u.String()).Return(&bundle.ServiceInstance{
+					ID: u,
+					Spec: &bundle.Spec{
+						Alpha: map[string]interface{}{
+							"dashboard_redirect": true,
+						},
+					},
+				}, nil)
+			},
+			dashboardURL: fmt.Sprintf("https://%v/?id=%v", "url.com", u.String()),
+		},
+		{
+			name: "get service instance with spec alpha no dashboard redirect",
+			dao:  new(mocks.Dao),
+			config: Config{
+				DashboardRedirector: "https://url.com",
+			},
+			addExpectations: func(d *mocks.Dao) {
+				d.On("GetServiceInstance", u.String()).Return(&bundle.ServiceInstance{
+					ID: u,
+					Spec: &bundle.Spec{
+						Alpha: map[string]interface{}{
+							"testing": true,
+						},
+					},
+				}, nil)
+			},
+		},
+		{
+			name:   "get service instance with spec alpha no dashboard redirect",
+			dao:    new(mocks.Dao),
+			config: Config{},
+			addExpectations: func(d *mocks.Dao) {
+				d.On("GetServiceInstance", u.String()).Return(&bundle.ServiceInstance{
+					ID: u,
+					Spec: &bundle.Spec{
+						Alpha: map[string]interface{}{
+							"testing": true,
+						},
+					},
+				}, nil)
+			},
+		},
+		{
+			name:   "get service instance with spec alpha dashboard redirect invalid value",
+			dao:    new(mocks.Dao),
+			config: Config{},
+			addExpectations: func(d *mocks.Dao) {
+				d.On("GetServiceInstance", u.String()).Return(&bundle.ServiceInstance{
+					ID: u,
+					Spec: &bundle.Spec{
+						Alpha: map[string]interface{}{
+							"dashboard_redirect": "invalid_value",
+						},
+					},
+				}, nil)
+			},
+		},
+		{
+			name:   "get service instance with spec alpha dashboard redirect false",
+			dao:    new(mocks.Dao),
+			config: Config{},
+			addExpectations: func(d *mocks.Dao) {
+				d.On("GetServiceInstance", u.String()).Return(&bundle.ServiceInstance{
+					ID: u,
+					Spec: &bundle.Spec{
+						Alpha: map[string]interface{}{
+							"dashboard_redirect": false,
+						},
+					},
+				}, nil)
+			},
+		},
+		{
+			name:   "get service instance error is not found",
+			dao:    new(mocks.Dao),
+			config: Config{},
+			addExpectations: func(d *mocks.Dao) {
+				d.On("GetServiceInstance", u.String()).Return(nil, unknownError)
```

Comment:
just trying to understand. Is this specifying the behavior when the GetServiceInstance is called or is it the expected result?

Reply:
I think the answer is both.

https://github.com/stretchr/testify#mock-package 

```go
// TestSomething is an example of how to use our test object to
// make assertions about some target code we are testing.
func TestSomething(t *testing.T) {

  // create an instance of our test object
  testObj := new(MyMockedObject)

  // setup expectations
  testObj.On("DoSomething", 123).Return(true, nil)

  // call the code we are testing
  targetFuncThatDoesSomethingWithObj(testObj)

  // assert that the expectations were met
  testObj.AssertExpectations(t)
```